doc/tutorial/conv_arithmetic.txt

```diff
@@ -0,0 +1,597 @@
+    o &= i + 2 \lfloor k / 2 \rfloor - (k - 1) \\
+      &= i + 2n - 2n \\
+      &= i.
+    \end{split}
+
+In other words, **the output size is equal to the input size**. This is
+sometimes referred to as ``half`` (or ``same``) padding. Here is an example for
+:math:`i = 5`, :math:`k = 3` and (therefore) :math:`p = 1`:
+
+.. figure:: conv_arithmetic_figures/same_padding_no_strides.gif
+
+.. todo::
+
+    * Tie this to Theano's and Theano-dependent packages' terminology.
+    * What happens if kernel size is even?
+
+Full
+^^^^
+
+If :math:`p = k - 1`, we have that
+
+.. math::
+
+    \begin{split}
+    o &= i + 2(k - 1) - (k - 1) \\
+      &= i + (k - 1).
+    \end{split}
+
+The input size is *increased* by :math:`k - 1` rather than decreased by
+:math:`k - 1`, as was the case for the non-padded convolution. This is
+sometimes referred to as ``full`` padding, because every element of the
+kernel is superimposed on every element of the input feature map. Here is an
+example for :math:`i = 5`, :math:`k = 3` and (therefore) :math:`p = 2`:
+
+.. figure:: conv_arithmetic_figures/full_padding_no_strides.gif
+
+Strides
+=======
+
+Up until now, we only considered convolutions where the kernel was translated by
+increments of :math:`s = 1`. Let's now examine what happens when :math:`s > 1`.
+To keep things simple, let's ignore padding for the moment, i.e.
+
+* input size :math:`i`,
+* kernel size :math:`k`,
+* stride :math:`s`,
+* padding size :math:`p = 0`.
+
+Here is what the strided convolution looks like for :math:`i = 5`, :math:`k = 3`
+and :math:`s = 2`:
+
+.. figure:: conv_arithmetic_figures/no_padding_strides.gif
+
+Once again, we are counting the number of positions where the kernel can be
+placed.  Let's consider the x-axis: the left side of the kernel starts at the
+left side of the input, but this time we slide the kernel **by hops of**
+:math:`s` until its right side touches the right side of the input. The size of
+the output will be equal to the number of steps we need to make, plus one,
+because we need to account for the initial position of the kernel. The same
+applies for the y-axis.
+
+From this, we conclude that the output size :math:`o` of the strided convolution
+is
+
+.. math::
+
+    o = \left\lfloor \frac{i - k}{s} \right\rfloor + 1.
+
+Combining padding and strides
+-----------------------------
+
+The most general case, i.e.
+
+* input size :math:`i`,
+* kernel size :math:`k`,
+* stride :math:`s`,
+* padding size :math:`p`,
+
+is obtained by applying both the rule we derived for strides and the one we
+derived for the padding. In other words,
+
+.. math::
+
+    o = \left\lfloor \frac{i + 2p - k}{s} \right\rfloor + 1.
+
+Here is what it looks like for :math:`i = 5`, :math:`k = 3`, :math:`s = 2` and
+:math:`p = 1`:
+
+.. figure:: conv_arithmetic_figures/padding_strides.gif
+
+Sometimes numbers are mean
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You may have noticed that the floor function in the general case formula will
```

Comment:
Here I would write that sometimes it is not possible to fully apply the kernel on the (padded) input so some parts are ignored. I would move the current sentence in a note, after the figure, or directly in the corresponding transposed convolution section, as this sentence is more interesting to understand the inverted convolution rather than for its forward application.